test/currencyConvertor.ts

```diff
@@ -1,73 +1,175 @@
+    await usdcContract.approve(
+    currencyConvertor.address,
+    1000e6,
+    );
   });
+
   describe("deposit", async () => {
-    it("deposit USDT to USDC", async () => {
-      await currencyConvertor.deposit(
-        usdc.address,
-        100,
+    it("deposit USDC to Uniswap", async () => {
+      const zeroExTransaction: any = await axiosRequest({
+        method: 'GET',
+        url: generateQueryPath(
+          'https://api.0x.org/swap/v1/quote',
+          {
+            sellToken: usdcAddres,
+            buyToken: uniswapAddress,
+            sellAmount: '1',
+            slippagePercentage: 1.0,
+          },
+        ),
+      });
+
+      const tx = await currencyConvertor.deposit(
+        usdcAddres,
+        '1000',
         zeroExExchangeWrapper.address,
-        100,
-        1,
-        0x4,
+        '10000', // starkKey
+        '100000', // positionId
+        encode(zeroExTransaction.to, zeroExTransaction.data),
       );
+
+      const blocks = await tx.wait();
+      const events = _.chain(blocks.events!)
+      .filter((e) => e.event === 'LogConvertedDeposit')
+      .value();
+
+      const event = events[0];
+      expect(event.args?.tokenFromAmount.toString()).to.equal('1000');
+      expect(event.args?.tokenFrom.toLowerCase()).to.equal(usdcAddres);
+    });
+
+    it("deposit USDC to Uniswap with invalid call data address", async () => {
+      const zeroExTransaction: any = await axiosRequest({
+        method: 'GET',
+        url: generateQueryPath(
+          'https://api.0x.org/swap/v1/quote',
+          {
+            sellToken: usdcAddres,
+            buyToken: uniswapAddress,
+            sellAmount: '1',
+            slippagePercentage: 1.0,
+          },
+        ),
+      });
+
+      try {
+        await currencyConvertor.deposit(
+          usdcAddres,
+          '1000',
+          zeroExExchangeWrapper.address,
+          '10000', // starkKey
+          '100000', // positionId
+          encode(zeroExTransaction.from, zeroExTransaction.data),
+        );
+      } catch (error) {
+        expect(error.reason).to.equal('invalid address (argument="address", value=undefined, code=INVALID_ARGUMENT, version=address/5.4.0)');
+      }
+    });
+
+    it("deposit USDC to Uniswap without enough funds", async () => {
+      const zeroExTransaction: any = await axiosRequest({
+        method: 'GET',
+        url: generateQueryPath(
+          'https://api.0x.org/swap/v1/quote',
+          {
+            sellToken: usdcAddres,
+            buyToken: uniswapAddress,
+            sellAmount: '10000000',
+            slippagePercentage: 1.0,
+          },
+        ),
+      });
+
+      try {
+        await currencyConvertor.deposit(
+          usdcAddres,
+          '1000',
+          zeroExExchangeWrapper.address,
+          '10000', // starkKey
+          '100000', // positionId
+          encode(zeroExTransaction.to, zeroExTransaction.data),
+        );
+      } catch (error) {
+      }
```

Comment:
Can you verify an error is thrown with this Chai matcher: https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#revert-with-message